app/src/main/java/com/yamato/mymvpandsqlite/UI/SignUp/SignUpActivity.kt

```diff
@@ -1,14 +1,79 @@
 package com.yamato.mymvpandsqlite.UI.SignUp
 
+import android.content.Intent
 import android.os.Bundle
 import android.os.PersistableBundle
+import android.view.View
+import android.widget.Button
+import android.widget.EditText
+import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
 import com.yamato.mymvpandsqlite.R
+import com.yamato.mymvpandsqlite.UI.Main.MainActivity
+import com.yamato.mymvpandsqlite.UI.SignIn.SignInActivity
 
 
-class SignUpActivity : AppCompatActivity(){
-    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
+class SignUpActivity : AppCompatActivity(), SignUpContact.View, View.OnClickListener {
+    private lateinit var mTextUsername : EditText
+    private lateinit var mTextEmail : EditText
+    private lateinit var mTextPassword : EditText
+    //tam thoi chi quan tam Username va Email de dang ki
+    private lateinit var mButtonSignIn : Button
+    private lateinit var mButtonSignUp : Button
+    private lateinit var mSignUpPresenter: SignUpPresenter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_sign_up)
+
+        initView()
+        registerListener()
+        initPresenter()
+    }
+
+    private fun initView(){
+        mTextUsername = findViewById(R.id.importUsername)
+        mTextPassword = findViewById(R.id.importPassword)
+        mTextEmail = findViewById(R.id.importEmail)
+        mButtonSignUp = findViewById(R.id.btnRegister)
+        mButtonSignIn = findViewById(R.id.btnBackToLogin)
+    }
+
+    private fun registerListener(){
+        mButtonSignIn.setOnClickListener(this)
+        mButtonSignUp.setOnClickListener(this)
+    }
+
+    private fun initPresenter(){
+        mSignUpPresenter = SignUpPresenter()
+        mSignUpPresenter.setView(this)
+    }
+
+    override fun signUpSuccess() {
+        Toast.makeText(this, "Sign Up Success !", Toast.LENGTH_LONG).show()
+        startActivity(Intent(this, MainActivity::class.java))
```

Comment:
Nên cho back về sign-in để người dùng đăng nhập

Reply:
Oke